### PR TITLE
Added FHIR parser functional tests.

### PR DESCRIPTION
**Pull request by ACMoretxj**
* Added FHIR parser functional tests, using all sample data as testing source.
* Removed `valueRange` property from `Resource/Observation` since it can't pass parser checking.
* Updated related test resources.

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Liquid.Converter/Microsoft.Health.Fhir.Liquid.Converter.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <PackageReference Include="Antlr4.Runtime.Standard" Version="4.8.0" />
     <PackageReference Include="DotLiquid" Version="2.0.366" />
+    <PackageReference Include="Hl7.Fhir.R4" Version="2.0.3" />
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.Liquid.Converter/Microsoft.Health.Fhir.Liquid.Converter.csproj:12]: Can we move the reference to the test project?

File: data/Templates/Hl7v2/Resource/_Observation.liquid [hunk 2]
```diff
@@ -58,24 +58,6 @@
                 {% endif -%}{% endif -%}
             },
         },
-        "valueRange":
```
Comment (moria97) [inline data/Templates/Hl7v2/Resource/_Observation.liquid:61]: Will this deleted section cause "parse errors"?

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs [hunk 3]
```diff
@@ -140,6 +142,24 @@ public async Task CheckPassOfficialValidator(string templateName, string sampleP
             Directory.Delete(resultFolder, true);
         }
 
+        [Theory]
+        [MemberData(nameof(GetHL7V2Cases))]
+        [MemberData(nameof(GetCCDACases))]
+        public async Task CheckPassFhirParser(string templateName, string samplePath)
```
Comment (moria97) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:148]: Better to add a sample that fails the parsing

**Comment by BoyaWu10** [comment]
The reason it cannot pass the parser check seems that the value should be an integer (`182`) instead of a string (`"182"`). Can we try to fix the template instead of remove it?

**Comment by ACMoretxj** [comment]
> The reason it cannot pass the parser check seems that the value should be an integer (`182`) instead of a string (`"182"`). Can we try to fix the template instead of remove it?

I've tried to do that before and it puts the folloing result, the weird point is the exception remains the same. So I decide to just remove the property currently and fix it later.
```json
"valueRange": {
      "low": {
            "value": 182
      }
},
```

File: data/Templates/Hl7v2/Resource/_Observation.liquid (hunk 2 above)
Comment (ACMoretxj) [reply data/Templates/Hl7v2/Resource/_Observation.liquid:61]: Yes, or there will be a `{"While building a POCO: Element 'value' must not repeat (at Bundle.entry[5].resource[0].valueRange[0])"}` error.

File: src/Microsoft.Health.Fhir.Liquid.Converter/Microsoft.Health.Fhir.Liquid.Converter.csproj (hunk 1 above)
Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.Liquid.Converter/Microsoft.Health.Fhir.Liquid.Converter.csproj:12]: Very reasonable, I'll do that.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs (hunk 3 above)
Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:148]: The original intention is to make sure our conversion result could pass the parser checking. If we add expect-fail-cases, it's more like we are testing the parser itself.
But in order ensure the basic functionality of the parser, I think it's also reasonable to add a basic expect-fail-case, I'll do that later.

Comment (moria97) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/RuleBasedTests.cs:148]: Yes. Just want to make sure "Bad templates" will fail the test:)

**Comment by BoyaWu10** [comment]
> > The reason it cannot pass the parser check seems that the value should be an integer (`182`) instead of a string (`"182"`). Can we try to fix the template instead of remove it?
> 
> I've tried to do that before and it puts the folloing result, the weird point is the exception remains the same. So I decide to just remove the property currently and fix it later.
> 
> ```json
> "valueRange": {
>       "low": {
>             "value": 182
>       }
> },
> ```

Sure. I saw the exception above and it is because `valueQuantity` and `valueRange` cannot appear at the same time, as defined in FHIR spec. We can fix it later.

**Review by BoyaWu10 — APPROVED** [review]

**Review by moria97 — APPROVED** [review]

**Pull request merged**